src/bionics_ui.cpp

```diff
@@ -510,175 +511,135 @@ void player::power_bionics()
+                    }
+                    break;
+                }
             }
             redraw = true;
-            const int newch = popup_getkey( _( "%s; enter new letter. Space to clear. Esc to cancel." ),
-                                            tmp->id->name );
-            wrefresh( wBio );
-            if( newch == ch || newch == KEY_ESCAPE ) {
-                continue;
-            }
-            if( newch == ' ' ) {
-                tmp->invlet = ' ';
-                continue;
-            }
-            if( !bionic_chars.valid( newch ) ) {
-                popup( _( "Invalid bionic letter. Only those characters are valid:\n\n%s" ),
-                       bionic_chars.get_allowed_chars() );
-                continue;
-            }
-            bionic *otmp = bionic_by_invlet( newch );
-            if( otmp != nullptr ) {
-                std::swap( tmp->invlet, otmp->invlet );
-            } else {
-                tmp->invlet = newch;
-            }
-            // TODO: show a message like when reassigning a key to an item?
         } else if( action == "NEXT_TAB" ) {
             redraw = true;
             scroll_position = 0;
             cursor = 0;
-            if( tab_mode == TAB_ACTIVE ) {
-                tab_mode = TAB_PASSIVE;
-            } else {
-                tab_mode = TAB_ACTIVE;
+            if( !current_bionic_list.empty() ) {
+                do {
+                    cur_tab_idx = ( cur_tab_idx + 1 ) % BionicsDisplayType::displayTypes.size();
+                } while( bionics_by_type[cur_tab_idx].empty() );
             }
         } else if( action == "PREV_TAB" ) {
             redraw = true;
             scroll_position = 0;
             cursor = 0;
-            if( tab_mode == TAB_PASSIVE ) {
-                tab_mode = TAB_ACTIVE;
-            } else {
-                tab_mode = TAB_PASSIVE;
+            if( !current_bionic_list.empty() ) {
+                do {
+                    if( --cur_tab_idx < 0 ) {
+                        cur_tab_idx = BionicsDisplayType::displayTypes.size() - 1;
+                    }
+                } while( bionics_by_type[cur_tab_idx].empty() );
             }
-        } else if( action == "REASSIGN" ) {
-            menu_mode = REASSIGNING;
-        } else if( action == "TOGGLE_EXAMINE" ) { // switches between activation and examination
-            menu_mode = menu_mode == ACTIVATING ? EXAMINING : ACTIVATING;
-            redraw = true;
         } else if( action == "HELP_KEYBINDINGS" ) {
             redraw = true;
         } else if( action == "CONFIRM" ) {
-            confirmCheck = true;
+            tmp = current_bionic_list[cursor];
+            need_activate = true;
+        } else if( action == "USAGE_HELP" ) {
+            std::string help_str;
+            for( size_t i = 0; i < column_num_entries; i++ ) {
+                bionic_col_data col_data = get_col_data( i );
+                help_str += colorize( col_data.get_name(), c_white );
```

Comment:
IMO one `help_str += string_format( ... );` will be better.

Reply:
BevapDin said exactly opposite in one of my previous PRs...